### PR TITLE
Add deleteDefinitionElements

### PR DESCRIPTION
**Pull request by scsewall**
Implement `deleteDefinitionElements` that will check for usage of DefinitionElements before deleting them.

### CODE REVIEW

File: core/backend/src/test/standalone/DeleteDefinitionElements.test.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      IModelJsFs.mkdirSync(outputDir);
+    }
+    // initialize logging
+    if (false) {
+      Logger.initializeToConsole();
+      Logger.setLevelDefault(LogLevel.Error);
+    }
+  });
+
+  it.only("should delete if not used", async () => {
```
Comment (pmconne) [inline core/backend/src/test/standalone/DeleteDefinitionElements.test.ts:34]: Remove `.only`.

File: core/backend/src/IModelDb.ts [hunk 2]
```diff
@@ -1615,25 +1616,33 @@ export namespace IModelDb { // eslint-disable-line no-redeclare
...
-          } finally {
-            this._iModel.nativeDb.endPurgeOperation();
+      let notDeletedIds: Id64Set = new Set<Id64String>();
+      if (usageInfo.usedIds) {
+        notDeletedIds = Id64.toIdSet(usageInfo.usedIds);
+      }
+      if (usageInfo.viewDefinitionIds) {
+        // take another pass in case a deleted ViewDefinition was the only usage of these view-related DefinitionElements
+        let viewRelatedIds: Id64Array = [];
+        if (usageInfo.displayStyleIds) { viewRelatedIds = viewRelatedIds.concat(usageInfo.displayStyleIds.filter((id) => usedIdSet?.has(id))); }
```
Comment (pmconne) [inline core/backend/src/IModelDb.ts:1626]: boring nitpick: formatting.

**Review by pmconne — CHANGES_REQUESTED** [review]

File: core/backend/src/IModelDb.ts (hunk 2 above)
Comment (scsewall) [reply core/backend/src/IModelDb.ts:1626]: Trying to keep this method from getting too long...

Comment (pmconne) [reply core/backend/src/IModelDb.ts:1626]: Long readable function better than short less-readable function containing longer lines IMO but not gonna twist your arm.

File: core/backend/src/test/standalone/DeleteDefinitionElements.test.ts (hunk 1 above)
Comment (scsewall) [reply core/backend/src/test/standalone/DeleteDefinitionElements.test.ts]: Removed .only since I am ready to publish this PR.

**Review by pmconne — APPROVED** [review]

**Pull request merged**